feat(ConfigProvider): Modify config merge to do replace instead of update for Specific tags for Remotely managed collectors (#1604)
SumoLogic/sumologic-otel-collector: docs/upgrading.md

For example, if two configuration files(say `conf1.yaml` and `conf2.yaml`) define the `collector_fields`,

In previous flow, the values of `collector_fields` from `conf2.yaml` will be added to `conf1.yaml` and the effective configuration will have vaules from both the configurations.

Starting from `v0.103.0-sumo-0`, the values of `collector_fields` tag from `conf1.yaml` will be overwritten by values from `conf2.yaml` and the effective configuration will have `collector_fields` value of `conf2.yaml` only.

For example:

`conf1.yaml`:

```
extensions:
  sumologic:
    collector_description: "My OpenTelemetry Collector"
    collector_fields:
      cluster: "cluster-1"
    some_list:
      - element 1
      - element 2
```

`conf2.yaml`:

```
extensions:
  sumologic:
    collector_fields:
      zone: "eu"
    some_list:
      - element 3
      - element 4
```

effective configuration (`old behaviour`)

```
extensions:
  sumologic:
    collector_description: "My OpenTelemetry Collector"
    collector_fields:
      cluster: "cluster-1"
      zone: "eu"
    some_list:
      - element 3
      - element 4
```

> effective configuration (`new behaviour`)

Reviewer: is there an issue describing the need for the new behavior? If so please link the issue to the PR. If its a Jira issue then let's maybe describe the requirement in the PR.

Author: https://docs.google.com/document/d/1U9yb2SAW3Q-c3zWimMwoFRAP0jsdGioijlbcUCHZ4wY

This change was done as part of this feature.

```
extensions:
  sumologic:
    collector_description: "My OpenTelemetry Collector"
    collector_fields:
      zone: "eu"
    some_list:
      - element 3
      - element 4
```

If you have multiple config files with `collector_fields` key specified, only the value from the last file(**alphabetically** sorted order) will be present in effective configuration.

> So, for locally managed collectors, Please don't maintain `collector_fields` key in multiple configuration files and move them to a single file.

Reviewer 2: ```suggestion
Due to above, avoid maintaining `collector_fields` in multiple configuration files and move them to a single file.
```

      compression: ""
  ```

> ## Upgrading to v0.103.0-sumo-0

Reviewer 2: should be in 0.104.0 section
